### PR TITLE
feat: add optional git hooks

### PR DESCRIPTION
**Pull request by aldur**
## Description

Add _optional_ Git hooks for those who don't like to push code only to find it rejected by the CI linter.

## Testing Information

Read the new `README.md` section. After installing the hooks, try making a modification that `cargo fmt` won't like. Try to commit it, or try running `.git/hooks/pre-commit-make-lint.sh`.

## Checklist:

- [X] I have performed a self-review of my code
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by cylewitruk** [comment]
I personally like this as a starting point. I did a deep-dive on this topic a while back for [clarity-wasm](https://github.com/stacks-network/clarity-wasm/pull/101), but I never followed through on it. There might be some inspiration there using `commitlint` (not necessarily `commitizen`, that's much more on the convenience-tooling side).

File: Makefile [hunk 1]
```diff
@@ -181,3 +181,13 @@ run-signer:
 	RUST_LOG=info SIGNER_SIGNER__DB_ENDPOINT="postgres://devenv:devenv@localhost:$$POSTGRES_PORT/signer" cargo run --bin signer -- -c ./signer/src/config/default.toml --migrate-db
 
 .PHONY: run-signer
+
+# Git hooks
+# ----------------------------------------------------
+
+install-git-hooks:
+	mkdir -p .git/hooks
+	ln -s ../../devenv/hooks/pre-commit-make-lint.sh .git/hooks/
```
Comment (cylewitruk) [inline Makefile:190]: Why is this `../../devenv`, shouldn't it just be `./devenv/etc` if the make commands are run from the repo root?

Comment (aldur) [reply Makefile:190]: The link needs to be relative to the target (this has bit me before :))

**Comment by aldur** [comment]
> I personally like this as a starting point. I did a deep-dive on this topic a while back for [clarity-wasm](https://github.com/stacks-network/clarity-wasm/pull/101), but I never followed through on it. There might be some inspiration there using `commitlint` (not necessarily `commitizen`, that's much more on the convenience-tooling side).

Thanks! Unless there's a strong feeling, I'd rather keep things simple and re-use our Makefile rather than introduce additional tools -- I have also used `precommit` in the past, but our Makefile is already there doing the heavy lifting.

**Comment by cylewitruk** [comment]
@aldur yeah it was just "fyi" -- i just have the one question before i approve this :100:

File: Makefile (hunk 1 above)
Comment (cylewitruk) [reply Makefile:190]: Gotcha :+1:

**Review by cylewitruk — APPROVED** [review]

**Pull request merged**